File: Task_2/scatter.py
import pandas as pd
import matplotlib.pyplot as plt

# Spécifiez le chemin vers votre fichier CSV
file_path = '/content/Housing.csv'

# Chargez le fichier CSV dans un DataFrame pandas
df = pd.read_csv(file_path)


# Afficher les premières lignes du DataFrame
print("Premières lignes:")
print(df.head())

# Création du graphique de dispersion
plt.scatter(df['area'], df['price'])
plt.xlabel('Area')
plt.ylabel('Price')
plt.title('Scatter plot of Area vs Price')
plt.show()

"""Le graphique de dispersion (scatter plot)  montre la relation entre les variables 'area' (superficie) et 'price' (prix). Voici une interprétation de ce graphique :

Observations Générales
Tendance Générale :

Il semble y avoir une tendance générale à la hausse entre la superficie (area) et le prix (price). En d'autres termes, à mesure que la superficie augmente, le prix tend également à augmenter. Cela est attendu puisque des propriétés plus grandes ont tendance à coûter plus cher.
Dispersion des Données :

Les points sont assez dispersés, indiquant une variabilité significative dans les prix pour une superficie donnée. Cela suggère que d'autres facteurs en plus de la superficie influencent le prix.
Concentration des Données :

Il y a une concentration de points dans les zones de superficie entre 2000 et 8000 et de prix entre 2e6 et 8e6. Cela peut indiquer que la plupart des propriétés dans cet échantillon se situent dans cette gamme de tailles et de prix.
Outliers :

Quelques points sont éloignés des autres, indiquant des outliers. Par exemple, il y a des propriétés avec des prix très élevés même pour des superficies similaires à d'autres propriétés moins chères.
Interprétation Spécifique
Clusters de Données : Il y a plusieurs clusters ou groupes visibles dans le graphique, ce qui pourrait indiquer des segments de marché différents ou des caractéristiques spécifiques des propriétés qui les rendent plus ou moins chères à taille égale.

Non-linéarité : La relation entre la superficie et le prix n'est pas strictement linéaire. Il y a des cas où des petites augmentations de la superficie conduisent à des augmentations significatives du prix, et d'autres où de grandes superficies n'entraînent qu'une augmentation modérée du prix.


En résumé, ce scatter plot montre une corrélation positive générale entre la superficie et le prix, mais aussi une grande variabilité qui suggère que d'autres facteurs influencent le prix des propriétés.
"""

File: Task_2/scatter.R

# Charger les bibliothèques nécessaires
library(readr)
library(ggplot2)

# Spécifier le chemin vers votre fichier CSV
file_path <- "/content/Housing.csv"

# Charger le fichier CSV dans un DataFrame R (data.frame)
df <- read_csv(file_path)

# Afficher les premières lignes du DataFrame pour vérifier le chargement
print("Premières lignes:")
print(head(df))


# Création du graphique de dispersion
ggplot(df, aes(x = area, y = price)) +
  geom_point() +
  labs(x = "Area", y = "Price", title = "Scatter plot of Area vs Price")

"""
Le deuxième graphique de dispersion (scatter plot) fourni montre également la relation entre les variables 'area' (superficie) en abscisse et 'price' (prix) en ordonnées. Voici une analyse détaillée de ce plot :

Observations Générales
Tendance Générale :

Comme dans le premier graphique, il semble y avoir une tendance générale où une augmentation de la superficie (area) est associée à une augmentation du prix (price). Cela indique une relation positive entre ces deux variables.
Dispersion des Données :

Les points sont assez dispersés, montrant une grande variabilité des prix pour des superficies similaires. Cela signifie qu'il y a d'autres facteurs en jeu qui influencent le prix en plus de la superficie.
Concentration des Données :

Une concentration notable de points se trouve dans les zones où la superficie est inférieure à 8000 et le prix est inférieur à 1e7. Cela pourrait indiquer que la majorité des propriétés de l'échantillon se situent dans cette gamme de taille et de prix.
Outliers :

Plusieurs outliers sont visibles dans le graphique, particulièrement pour des superficies comprises entre 8000 et 16000, et des prix dépassant 1e7. Ces points indiquent des propriétés dont les prix sont beaucoup plus élevés par rapport à leur superficie ou vice versa.
Interprétation Spécifique
Clusters de Données : Il y a des groupes de données plus denses vers le bas du graphique (superficie entre 2000 et 8000, prix entre 5e6 et 1e7) ce qui pourrait suggérer des segments de marché spécifiques ou des caractéristiques communes des propriétés dans ces gammes.

Non-linéarité : La relation entre la superficie et le prix ne semble pas strictement linéaire. Il y a des variations où certaines propriétés avec des superficies plus grandes n'entraînent pas nécessairement une augmentation proportionnelle du prix.

Analyse Potentielle
Pour une analyse plus approfondie de la relation entre la superficie et le prix, vous pourriez :

Ajouter une ligne de tendance : Utiliser une régression linéaire ou non-linéaire pour visualiser la tendance générale et mieux comprendre la relation entre la superficie et le prix.

Segmenter par d'autres variables : Par exemple, inclure des variables telles que la localisation, l'âge des propriétés, le nombre de chambres, etc., pour comprendre mieux les facteurs influençant les prix.

Étudier les outliers : Examiner les propriétés qui sont des outliers pour identifier pourquoi elles sont si différentes du reste. Cela pourrait inclure des facteurs comme des rénovations récentes, des emplacements particulièrement prisés, ou des caractéristiques uniques.

Conclusion
Ce scatter plot montre une relation positive entre la superficie et le prix des propriétés, mais aussi une grande variabilité qui suggère l'influence de nombreux autres facteurs. En ajoutant des lignes de tendance et en stratifiant par d'autres variables, vous pourriez obtenir une compréhension plus nuancée de la relation entre ces deux variables.





"""